pkg/controller/snapshot_controller_base.go

```diff
@@ -275,7 +272,7 @@ func (ctrl *csiSnapshotController) contentWorker() {
 			// Skip update if content is for another CSI driver
 			snapshotClassName := content.Spec.VolumeSnapshotClassName
 			if len(snapshotClassName) != 0 {
-				if snapshotClass, err := ctrl.clientset.VolumesnapshotV1alpha1().VolumeSnapshotClasses().Get(snapshotClassName, metav1.GetOptions{}); err == nil {
+				if snapshotClass, err := ctrl.classLister.Get(snapshotClassName); err == nil {
```

Comment:
we can use the similar way of processing snapshot to process content. If snapshotClassName is nil, set default one. Otherwise try to find the class specified by user. If snapshotClass's snapshotter matches controller's snapshotter, it should process the object.